base_partner_sequence/models/partner.py

```diff
@@ -72,3 +74,21 @@ def _commercial_fields(self):
         to the partner's contacts
         """
         return super()._commercial_fields() + ["ref"]
+
+    def update_partners_ref(self):
+        """Calculate ref of all partners. Called from a server action"""
+        offset = 0
+
+        while True:
+            partners = self.search(
+                [("ref", "=", False)], offset=offset, limit=BATCH_SIZE
+            )
+
+            if not partners:
+                break
+
+            for partner in partners:
+                if partner._needs_ref():
+                    partner.write({})
```

Comment:
This is kind of weird.
I think I'd do this instead, which is more explicit, to-the-point and probably better performance-wise:

```suggestion
                partner.ref = partner._get_next_ref()
```